modules/catalog/src/main/java/org/apache/ignite/internal/catalog/storage/MakeIndexAvailableEntry.java

```diff
@@ -72,16 +73,60 @@ public CatalogEvent eventType() {
 
     @Override
     public CatalogEventParameters createEventParameters(long causalityToken, int catalogVersion) {
-        return new MakeIndexAvailableEventParameters(causalityToken, catalogVersion, descriptor.id());
+        return new MakeIndexAvailableEventParameters(causalityToken, catalogVersion, indexId);
+    }
+
+    private static CatalogSchemaDescriptor schemaByIndexId(Catalog catalog, int indexId) {
+        CatalogIndexDescriptor index = catalog.index(indexId);
+
+        assert index != null : indexId;
+
+        CatalogTableDescriptor table = catalog.table(index.tableId());
+
+        assert table != null : index.tableId();
+
+        CatalogSchemaDescriptor schema = catalog.schema(table.schemaId());
+
+        assert schema != null : table.schemaId();
+
+        return schema;
     }
 
-    /** Returns schema name. */
-    public String schemaName() {
-        return schemaName;
+    private static CatalogIndexDescriptor createReadWriteIndex(CatalogIndexDescriptor source, long causalityToken) {
```

Comment:
Again, 2 terminologies. Writing in some documentation that 'RW is same as Available' will not be enough if all around the code we see either Available, or RW. My suggestion is to pick 1 principal terminology (I suggest 'Registered/Available') and write the new code in accordance with it, and then change all the terminology in accordance with this one (for the code that was written earlier).

Reply:
Tried renaming everything related to this problem.